02_eval/eval.py

```diff
@@ -74,8 +73,45 @@ def load_dataset(args):
         raise ValueError(f"Unknown dataset: {args.data}")
 
 
-def load_model(args):
-    return load_lamini_model(args.model)
+class LaminiModelStage(GenerationNode):
+    def __init__(self, dataset, model_name="meta-llama/Meta-Llama-3-8B-Instruct",):
+        super().__init__(
+            model_name=model_name,
+            max_new_tokens=150,
+        )
+        self.model_name = model_name
+        self.dataset = dataset
+
+    def generate(
+        self,
+        prompt: Union[Iterator[PromptObject], AsyncIterator[PromptObject]],
+        *args,
+        **kwargs,
+    ):
+        results = super().generate(
+            prompt,
+            output_type=self.dataset.get_output_type(),
+            *args,
+            **kwargs,
+        )
+
+        return results
```

Comment:
We can remove the __init__ and generate() functions

for generate we can pass output_type in the pipeline like this
```
def forward(self, x):
        x = self.question_generator(
            x,
            output_type=dataset.get_output_type()
        )
        x = self.answer_generator(x)
        return x
        
      ```
      
      See below for how to change __init__

Reply:
Done
Moved the LaminiModelStage to eval_pipeline.py to be close to other subclasses of GenerationNode
Also rewrote ModifyStage ScoreStage to use its designed pattern